### PR TITLE
[FAN-362] Set up lint-staged

### PR DESCRIPTION
**Pull request by BigBadAlien**
https://fangible.atlassian.net/browse/FAN-362

## Follow the checklist to proceed

- [x] I've performed a code self-review
- [x] I've tested feature or bugfix on my own
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've checked mobile & tablet screens (if have changes in UI)
- [x] Local build successfully completed
- [x] Branch name and pull request title are according to [the requirements](https://fangible.atlassian.net/wiki/spaces/FAN/pages/62226433/Branches+naming)

### CODE REVIEW

**Comment by BigBadAlien** [comment]
A way for review:

```
git checkout feature/FAN-362-lint-staged
rm -rf node_modules
yarn install
```

Change any ts or tsx file

```
git add -u
git commit -m 'Foobar'
```

The console output should be like this

```
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...
```

**Review by Denise54321 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -83,7 +83,8 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "format": "prettier --write \"**/*.{js,jsx,ts,tsx}\"",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "lint-staged": "lint-staged"
```
Comment (Antroll) [inline package.json:86]: I think it's not necessary. It's just a task for launching lint-staged itself. But what is the reason to use it?

File: .husky/pre-commit [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-yarn lint-staged
+npm run-script lint-staged
```
Comment (Antroll) [inline .husky/pre-commit]:
```suggestion
yarn run lint-staged
```
If you want to do that more specific.
We need to use `yarn` to be more consistent.

**Review by Antroll — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (BigBadAlien) [reply package.json:86]: Trying to resolve executaion permissions issue

Comment (BigBadAlien) [reply package.json:86]: Screenshot 2021-07-01 at 13 02 30 (1)

**Pull request merged**